src/protocol/command.rs

```diff
@@ -224,6 +282,86 @@ mod command {
+                let resolved = Command::executable_full_path(&command.executable);
+                assert!(resolved.is_absolute()); // assert path begins with /
+                Ok(())
+            }
+
+            #[test]
+            fn resolves_executable_with_full_path_to_itself() -> R<()> {
+                let command = Command::new("/bin/cp")?;
+                let resolved = Command::executable_full_path(&command.executable);
+                assert_eq!(
+                    resolved.as_os_str().as_bytes(),
+                    command.executable.as_slice()
+                );
+                Ok(())
+            }
+
+            #[test]
+            fn resolves_executable_with_relative_path_to_itself() -> R<()> {
+                let command = Command::new("./bin/cp")?;
+                let resolved = Command::executable_full_path(&command.executable);
+                assert_eq!(
+                    resolved.as_os_str().as_bytes(),
+                    command.executable.as_slice()
+                );
+                Ok(())
+            }
+
+            #[test]
+            fn gets_file_name_of_executable_in_path() -> R<()> {
+                let command = Command::new("cp")?;
+                let resolved = Command::executable_full_path(&command.executable);
+                let file_name = Command::executable_file_name(resolved.as_os_str().as_bytes());
+                assert_eq!(file_name, "cp");
+                Ok(())
+            }
+
+            #[test]
+            fn shortens_executable_with_relative_path_to_itself() -> R<()> {
+                let command = Command::new("./bin/cp")?;
+                let file_name = Command::executable_file_name(&command.executable);
+                assert_eq!(file_name, OsString::from("./bin/cp"));
+                Ok(())
+            }
+
+            #[test]
+            fn shortens_executable_that_is_not_in_path_to_itself() -> R<()> {
+                let command = Command::new("/share/doesnotexist")?;
+                let file_name = Command::executable_file_name(&command.executable);
+                assert_eq!(file_name, OsString::from("/share/doesnotexist"));
+                Ok(())
+            }
+
+            #[test]
+            fn updates_command_with_full_path_of_executable() -> R<()> {
+                let command = Command::new("cp")?;
+                let resolved = command.resolve_executable_path();
+                assert!(Command::path_from_bytes(&resolved.executable).is_absolute());
+                Ok(())
+            }
+
+            #[test]
+            fn updates_command_with_shortened_exeucutable_if_in_path() -> R<()> {
+                let command = Command::new("cp")?;
+                assert_eq!(
+                    command
+                        .resolve_executable_path()
+                        .shorten_executable_in_path(),
+                    command
+                );
+                Ok(())
+            }
```

Comment:
I would try to separate the tests that test `resolve_executable_path` from the ones that test `executable_file_name`. Into their own submodules.

Reply:
Ok; I moved tests into a separate module for each method under test.